lib/ansible/plugins/action/validate_argument_spec.py

```diff
@@ -77,6 +109,44 @@ def run(self, tmp=None, task_vars=None):
         if not isinstance(provided_arguments, dict):
             raise AnsibleError('Incorrect type for provided_arguments, expected dict and got %s' % type(provided_arguments))
 
+        if self._task.implicit:
+
+            # types handled by the argument parser
+            option_types = [None, 'bits', 'bool', 'bytes', 'dict', 'float', 'int', 'json', 'jsonarg', 'list', 'path', 'raw', 'sid', 'str']
+            meta_validator = ArgumentSpecValidator(
+                dict(
+                    description=dict(),
+                    type=dict(choices=option_types),
+                    default=dict(),
+                    choices=dict(type='list'),
+                    elements=dict(choices=option_types),
+                    options=dict(type='dict'),
+                    required=dict(type='bool'),
+                    # don't deter documentation, the argparser will ignore these
+                    version_added=dict(),
+                    version_added_collection=dict(),
```

Comment:
Wondering what others would think about including these two. It was so flexible before, that I worry the warnings might be noisy/unhelpful in some cases.

Reply:
If this is only about role argument specs, I don't think `version_added_collection` makes any sense, since options cannot come from other collections. (There are no doc fragments for role argument specs.)

`version_added` has been supported by antsibull-docs since the beginning.

Reply:
It would only have occurred to me to use `version_added` for standalone roles... are you using collection versions in the `version_added` field?